torch/csrc/utils/tensor_new.cpp

```diff
@@ -251,7 +251,7 @@ Tensor internal_new_from_data(
     // infer the scalar type and device type; it's not expected to infer the layout since these constructors
     // are defined per-layout-type (e.g. tensor vs sparse_coo_tensor).
     const auto& inferred_scalar_type = type_inference ? var.scalar_type() : scalar_type;
-    auto device = device_opt.has_value() ? *device_opt : (type_inference ? var.device() : at::Device(computeDeviceType(dispatch_key)));
+    auto device = device_opt.has_value() ? *device_opt : (type_inference ? var.device() : at::Device(computeDeviceType(dispatch_key.first), dispatch_key.second));
```

Comment:
This is the existing line that needs to change to no longer cause dtype to have an affect on device, right? We really don't want there to be any connection between dtype and device.

Reply:
Sure, but it can be solved by using the right `dispatch_key`.  For now, I will not touch this `internal_new_from_data`  function.